### PR TITLE
Capture exception snapshots only once an hour

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Keep a timestamp associated to the fingerprint to know when last time we capture snapshots for this exception. Hardcoded once an hour will probably make it a config token for this later

# Motivation

# Additional Notes

Jira ticket: [DEBUG-2068]

[DEBUG-2068]: https://datadoghq.atlassian.net/browse/DEBUG-2068?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeInstrumentationTest.java [hunk 1]
```diff
@@ -202,6 +204,35 @@ public void recursive() throws Exception {
     assertEquals("10", getValue(snapshot9.getCaptures().getReturn().getArguments().get("n")));
   }
 
+  @Test
+  public void captureOncePerHour() throws Exception {
+    Config config = createConfig();
+    Clock clockMock = mock(Clock.class);
+    when(clockMock.instant()).thenReturn(Instant.now());
```
Comment (ojung) [inline dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeInstrumentationTest.java:211]: When using mockito to mock clocks I got weird flaky tests in the past about unfinished mockings. Have you ever had that? I decided to have a method with two signatures, one with clock and the other one without the clock using a "default clock" to fix it.

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java [hunk 2]
```diff
@@ -59,7 +75,7 @@ public void createProbesForException(String fingerprint, StackTraceElement[] sta
   }
 
   void addFingerprint(String fingerprint) {
-    fingerprints.add(fingerprint);
+    fingerprints.put(fingerprint, Instant.ofEpochSecond(0));
```
Comment (ojung) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java:78]: There is also `Instant.MIN`: https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#MIN

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java [hunk 3]
```diff
@@ -70,15 +86,24 @@ private static ExceptionProbe createMethodProbe(
...
-    return fingerprints.contains(fingerprint);
+    return shouldCaptureException(fingerprint, clock);
+  }
+
+  boolean shouldCaptureException(String fingerprint, Clock clock) {
+    Instant lastCapture = fingerprints.get(fingerprint);
+    if (lastCapture == null) {
+      return false;
+    }
+    // only capture once an hour
```
Comment (ojung) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java:105]:
```suggestion
    // only capture once a captureInterval
```

Comment (ojung) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java:105]: Or delete the comment all together.

File: dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeInstrumentationTest.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/test/java/com/datadog/debugger/exception/ExceptionProbeInstrumentationTest.java:211]: no problem with mocking so far 🤞

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java (hunk 3 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java:105]: done

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java (hunk 2 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/exception/ExceptionProbeManager.java:78]: done, thanks 👌

**Pull request merged**